Sources_v3/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -294,6 +294,25 @@ class MessageUpdater<ExtraData: ExtraDataTypes>: Worker {
             attachmentDTO.localState = .pendingUpload
         }, completion: completion)
     }
+
+    /// Updates local state of attachment with provided `id` to be enqueued by attachment uploader.
+    /// - Parameters:
+    ///   - id: The attachment identifier.
+    ///   - completion: Called when the attachment database entity is updated. Called with `Error` if update fails.
+    func resendMessage(with messageId: MessageId, completion: @escaping (Error?) -> Void) {
+        database.write({
+            let messageDTO = try $0.messageEditableByCurrentUser(messageId)
+
+            guard messageDTO.localMessageState == .sendingFailed else {
+                throw ClientError.MessageEditing(
+                    messageId: messageId,
+                    reason: "message is in `\(messageDTO.localMessageState!)` state"
```

Comment:
This error message is is not helpful at all. `"Only messages with sendingFailed local state can be resent. The message state is \(messageDTO.localMessageState!)"`

Reply:
changed in ed6cf54d6967cb31422b2ad952a39e052cb855e4